### ISSUE TITLE
scancode job will be killed

### ISSUE THREAD

**Issue by AudiAlex** [fossology/fossology#2642]
### Description

scancode job is not able to finish and job will be killed.

Log:
2023-11-27 12:57:56 scancode [0] :: JOB[36].scancode[1528.localhost]: agent has not set the alive flag in at least 10 minutes, killing
2023-11-27 12:57:57 scancode [0] :: JOB[36].scancode[1528.localhost]: agent was killed by signal: 9.Killed

#### How to reproduce

I just startet a job with the scancode options

#### Screenshots

![image](https://github.com/fossology/fossology/assets/130551143/fa853787-da9b-4eca-aead-6433667d5eda)

### Versions

* Last commit id on master:Version: [4.4.0.0], Branch: [master], Commit: [#7ba0bc] 2023/11/22 09:23 +00:00 built @ 2023/11/22 09:40 +00:00
* Docker image

**Comment by uholeschak** [comment]
This is easy to fix. Change this line in scancode.conf:
`special[] = NOKILL`

**Comment by alpianon** [comment]
After upgrading to 4.4.0 (installation from sources)I encountered the same issue, but randomly: it happens roughly on 1 out of 10 uploads or so (both big and small, so upload size doesn't seem to matter). I applied the suggested fix, and now it seem to work. I guess that this issue should be investigated a little bit more. What could I do to provide more debug info?

**Comment by uholeschak** [comment]
Use my patch above and it will not be killed any more.

**Comment by alpianon** [comment]
yes @uholeschak but IIUC this means that the scancode agent does not do something that it is supposed to do, or that something else wrong is happening so that the sceduler believes the agent is hanging and needs to be killed. So your workaround works but IMHO it would be better to find and fix the issue

**Comment by uholeschak** [comment]
No this is not a bug, simply your system is to slow and a timeout occurs.

**Comment by alpianon** [comment]
> Use my patch above and it will not be killed any more. I am using now an automated script that patches the docker image after installation (with multiple fixes).

could you share the script? It would be super-useful! Thanks :slightly_smiling_face:

**Comment by uholeschak** [comment]
Basically the bat is like this (only the scancode patch in this case):
The chmod fixes an additional rights problem, 
```
SETLOCAL EnableDelayedExpansion

set "BatDir=%~dp0"
set "ContainerId=%1"

IF "!ContainerId!"=="" (
  echo missing container id
  EXIT /b 1
)

docker exec -i "!ContainerId!" sh -c "chmod a+xrw -R /root" || EXIT /b 1

docker cp "!BatDir!\scancode.conf.diff" "!ContainerId!:/usr/local/etc/fossology/mods-enabled/scancode/scancode.conf.diff" || EXIT /b 1
docker exec -i "!ContainerId!" sh -c "cd /usr/local/etc/fossology/mods-enabled/scancode && patch -N < /usr/local/etc/fossology/mods-enabled/scancode/scancode.conf.diff"
```
scancode.conf.diff:
```
--- scancode.conf	2023-11-22 10:38:18.000000000 +0100
+++ scancode_mod.conf	2023-12-08 17:52:33.757401300 +0100
@@ -20,4 +20,4 @@
 ; A comma separated list of values.
 ; Directives:
 ;     EXCLUSIVE: the agent cannot run concurrently with any other agent.
-special[] =
+special[] = NOKILL
```
